tensorrt_llm/_torch/auto_deploy/transformations/library/rope.py

```diff
@@ -1,6 +1,53 @@
+"""
+This transformation defines two main RoPE (Rotary Positional Embedding) pattern matchers used
+to identify and replace RoPE subgraphs with a custom op (`torch.ops.rope.flashinfer`).
+
+Supported RoPE variants:
+
+1. Explicit Cos/Sin Multiplication (HF-style, e.g., LLaMA, Mixtral, Qwen)
+   - Input layout: non-interleaved, [B, N, S, D] with unsqueeze_dim=1 and
+        [B, S, N, D] with unsqueeze_dim=2, default is [B, N, S, D]
+   - Frequencies are provided as separate `cos` and `sin` tensors of shape [B, S, head_dim].
+   - Source code:
+        def rotate_half(x):
+            x1 = x[..., : x.shape[-1] // 2]
+            x2 = x[..., x.shape[-1] // 2 :]
+            return torch.cat((-x2, x1), dim=-1)
+
+        def apply_rotary_pos_emb(q, k, cos, sin, position_ids=None, unsqueeze_dim=1):
```

Comment:
Let's add a custom op for this in `torch_rope.py` that is just reproducing the torch implementation similar to `apply_rope_ds`